### PR TITLE
Enable persistent queue in the build by default

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
**Description:**
Persistent Queue is a feature that currently requires "enable_unstable" tag provided to the the build to enable it. We have been using it in [our distro](https://github.com/SumoLogic/sumologic-otel-collector/) for several months already and it looks stable. It seems that others are using this capability as well (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9327#issuecomment-1135494065)

This change enables it by default, which also simplifies the build process a little bit.

This PR simply moves `queued_retry_experimental.go` to `queued_retry_inmemory.go` to make it easier to see the changes. This can be further cleaned up - the contents of `queued_retry_inmemory.go` should go into `queued_retry.go`, and the same with their respective tests. I'm planning to submit a separate PR with the cleanup to make this one easier to review.

This was originally part of #5711, but was then split into multiple PRs for clarity. #5764 and #5784 were already merged, and this is the final change.

**Link to tracking Issue:** N/A

**Documentation:** Changed the queue status from experimental to alpha.

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
> If the amount of changes makes this hard to understand, I can just submit the build changes and deletion of queued_retry_inmemory.go, and do the rest in a separate PR.

Probably a good idea

**Comment by bogdandrutu** [comment]
You should delete the experimental not the inmemory

**Comment by swiatekm-sumo** [comment]
> You should delete the experimental not the inmemory

Why not just copy the code to `queued_retry.go` in that case? As far as I know, the only reason it's not there to begin with is the in-memory/experimental split. I wanted to do that as a follow-up PR to this one, but I'm good either way.

**Comment by bogdandrutu** [comment]
> Why not just copy the code to queued_retry.go in that case? As far as I know, the only reason it's not there to begin with is the in-memory/experimental split. I wanted to do that as a follow-up PR to this one, but I'm good either way.

Because this way simplifies the reviewer job, and can easily see what changes, everything that is added in the inmemory which is right now used.

**Comment by swiatekm-sumo** [comment]
> > Why not just copy the code to queued_retry.go in that case? As far as I know, the only reason it's not there to begin with is the in-memory/experimental split. I wanted to do that as a follow-up PR to this one, but I'm good either way.
> 
> Because this way simplifies the reviewer job, and can easily see what changes, everything that is added in the inmemory which is right now used.

Ok, I see what you mean after having done it.

File: exporter/exporterhelper/queued_retry_experimental_test.go [hunk 1]
```diff
@@ -12,9 +12,6 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build enable_unstable
-// +build enable_unstable
-
 package exporterhelper
```
[the remark this answers is not in the document]
Comment (swiatekm-sumo) [reply exporter/exporterhelper/queued_retry_experimental_test.go:15]: I'm going to submit a PR merging these after this one is merged, don't think we need an issue for it.

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
```
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector/opentelemetry-collector/cmd/otelcorecol'
level=warning msg="[linters context] structcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649."
exporter/exporterhelper/queued_retry_inmemory.go:156: File is not `gofmt`-ed with `-s` (gofmt)
```

**Review by TylerHelmuth — APPROVED** [review]
LGTM

**Comment by TylerHelmuth** [comment]
@swiatekm-sumo if you need it I can help out with some of the new issues that were asked to be created.

**Comment by swiatekm-sumo** [comment]
> @swiatekm-sumo if you need it I can help out with some of the new issues that were asked to be created.

I think I'm going to be fine. I'll create the `configstorage` issue, and the rest will just be relatively simple PRs.

**Pull request merged**